src/intrinsics/ecma262/Object.js

```diff
@@ -413,30 +417,66 @@ export default function(realm: Realm): NativeFunctionValue {
   });
 
   // ECMA262 9.1.2.16
-  if (!realm.isCompatibleWith(realm.MOBILE_JSC_VERSION) && !realm.isCompatibleWith("mobile"))
-    func.defineNativeMethod("values", 1, (context, [O]) => {
+  if (!realm.isCompatibleWith(realm.MOBILE_JSC_VERSION) && !realm.isCompatibleWith("mobile")) {
+    let objectValues = func.defineNativeMethod("values", 1, (context, [O]) => {
       // 1. Let obj be ? ToObject(O).
       let obj = To.ToObject(realm, O);
 
+      // If we're in pure scope and the items are completely abstract,
+      // then create an abstract temporal with an array kind
+      if (realm.isInPureScope() && obj instanceof AbstractObjectValue) {
+        let array = ArrayValue.createTemporalWithWidenedNumericProperty(
+          realm,
+          [objectValues, obj],
+          ([methodNode, objNode]) => t.callExpression(methodNode, [objNode])
+        );
+        return array;
+      } else if (ArrayValue.isIntrinsicAndHasWidenedNumericProperty(obj)) {
```

Comment:
Could/should this happen in a non pure scope?

Reply:
The others will need to be changed to pure scope too.